### PR TITLE
Adding the aggregated operations.

### PR DESCRIPTION
**Pull request by aruniiird**
Aggregated operations like READ_WRITE_OPS, LOCK_OPS, INODE_OPS and
ENTRY_OPS are added to the metrics

Fixes:  gluster/gluster-prometheus/issues/98

Signed-off-by: Arun Kumar Mohan <arun.iiird@gmail.com>

### CODE REVIEW

**Comment by aruniiird** [comment]
Seems like all checks have passed
@sidharthanup @shtripat @aravindavk , @anmolsachan please review

**Comment by shtripat** [comment]
Can you send the fixes for profileinfo_gd2.go in separate PR?. Its nothing to do with this issues.

File: gluster-exporter/metric_volume.go [hunk 1]
```diff
@@ -173,8 +175,132 @@ var (
 		LongHelp:  "",
 		Labels:    volumeProfileFopInfoLabels,
 	})
+
+	glusterVolumeProfileFopTotalHitsAggregatedOps = newPrometheusGaugeVec(Metric{
```
Comment (shtripat) [inline gluster-exporter/metric_volume.go]: Make its fmt compliant

File: gluster-exporter/metric_volume.go [hunk 2]
```diff
@@ -173,8 +175,132 @@ var (
...
+		Namespace: "gluster",
+		Name:      "volume_profile_fop_total_hits_on_aggregated_fops",
```
Comment (shtripat) [inline gluster-exporter/metric_volume.go]: Make this name to represent that this is cumulative metrics as doing at line#190

File: gluster-exporter/metric_volume.go [hunk 3]
```diff
@@ -173,8 +175,132 @@ var (
...
+		Help: "Interval based total hits on aggregated FOPs" +
+			" like READ_WRIET_OPS, LOCK_OPS, INODE_OPS etc",
+		LongHelp: "",
+		Labels:   volumeProfileFopInfoLabels,
+	})
 )
 
+// OpType represents aggregated operations like
+// READ_WRITE_OPS, INODE_OPS, ENTRY_OPS, LOCK_OPS etc...
+type OpType struct {
```
Comment (shtripat) [inline gluster-exporter/metric_volume.go]: Do you need these functions to be public at all? Are they used anywhere else?

**Comment by aruniiird** [comment]
> Can you send the fixes for profileinfo_gd2.go in separate PR?. Its nothing to do with this issues.
@shtripat , acknowledged.
Created a new PR: https://github.com/gluster/gluster-prometheus/pull/113

File: gluster-exporter/metric_volume.go (hunk 1 above)
Comment (aruniiird) [reply gluster-exporter/metric_volume.go]: Already in go-fmt...

File: gluster-exporter/metric_volume.go (hunk 2 above)
Comment (aruniiird) [reply gluster-exporter/metric_volume.go]: As discussed, currently used the existing naming conventions, followed in the existing code, and probably, at a later point, we can do a refactoring...

File: gluster-exporter/metric_volume.go (hunk 3 above)
Comment (aruniiird) [reply gluster-exporter/metric_volume.go]: Don't require them to be public. Only reasoning was for a future case when we move them to a separate package, we have to just copy paste the code as is to the new place.

As discussed, will lower case the names.

**Comment by aravindavk** [comment]
Sorry, I missed a comment before merge. Please generate documentation using `make metrics-docgen` and send PR for the changes

**Pull request merged**